internal/controller/bucket_controller.go

```diff
@@ -808,3 +811,48 @@ func fetchIndexFiles(ctx context.Context, provider BucketProvider, obj *bucketv1
 
 	return nil
 }
+
+// sanitizeErrorMessage extracts all URLs from the error message
+// and replaces them with the URL without the query string.
+func sanitizeErrorMessage(errorMessage string) string {
```

Comment:
Can you replace with 

```go
type sanitizedError struct {
	err string
}

func (e sanitizedError) Error() string {
	return e.err
}


// sanitizeErrorMessage extracts all URLs from the error message
// and replaces them with the URL without the query string.
func sanitizeError(errorMessage string) sanitizedError {}
```

We could then wrap the error like:
```go
return fmt.Errorf("failed to get Etag for '%s' object: %w", sourceignore.IgnoreFile, sanitizeError(err.Error()))
```

Reply:
`sanitizeError` is confusing as the function takes a string not an error type.